### ISSUE TITLE
[Version ^3.0.0-alpha.2]  Cannot read property 'resolve' of undefined with Vue3 in SSR mode

### ISSUE THREAD

**Issue by oom-** [nuxt/vue-meta#718]
Hello, 

I'm trying to make work vue-meta with Vue3 in SSR mode (https://github.com/nuxt/vue-meta/tree/next#ssr)

Everything `build` just fine, I can start my express server and all but when I'm accessing a page I'm getting this error: 
```
[Vue warn]: Unhandled error during execution of setup function
  at <App>
(node:24140) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'resolve' of undefined
    at Object.resolve (F:\Documents\website\node_modules\vue-meta\dist\vue-meta.cjs.js:845:25)
    at recompute (F:\Documents\website\node_modules\vue-meta\dist\vue-meta.cjs.js:274:32)
    at compute (F:\Documents\website\node_modules\vue-meta\dist\vue-meta.cjs.js:434:27)
    at Object.addSource (F:\Documents\website\node_modules\vue-meta\dist\vue-meta.cjs.js:441:17)
    at MetaManager.addMeta (F:\Documents\website\node_modules\vue-meta\dist\vue-meta.cjs.js:757:34)
    at useMeta (F:\Documents\website\node_modules\vue-meta\dist\vue-meta.cjs.js:691:31)
    at setup (F:\Documents\website\dist\server\js\app.514800de.js:1622:42)
    at callWithErrorHandling (F:\Documents\website\node_modules\@vue\runtime-core\dist\runtime-core.cjs.js:6919:22)
    at setupStatefulComponent (F:\Documents\website\node_modules\@vue\runtime-core\dist\runtime-core.cjs.js:6537:29)
    at setupComponent (F:\Documents\website\node_modules\@vue\runtime-core\dist\runtime-core.cjs.js:6493:11)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:24140) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:24140) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This line trigger the error in `vue-meta.cjs.js`: 
![image](https://user-images.githubusercontent.com/11071671/131386910-5a190c07-43f0-4ef0-b155-c56eeb26af4a.png)

I tried to add the `metaManager` in the context (in case of): 
![image](https://user-images.githubusercontent.com/11071671/131387240-59b7bb55-db26-4e64-9e63-3b9cf4e9d507.png)

Inside the `createApp` function there is nothing special : 
![image](https://user-images.githubusercontent.com/11071671/131387379-1fbfac61-1ce7-4810-bf82-7cbd2c90ec64.png)

And here is my `App.vue` file where the error is triggered : 
![image](https://user-images.githubusercontent.com/11071671/131387501-d051449f-caf0-47c8-9bdc-bec68f07a178.png)

(Do not pay attention to the extra `console.log` I was checking that nothing was undefined.)

Does someone have an idea about what is going wrong here? 😕

Thank you.

**Comment by alowave223** [comment]
bump

**Comment by Nalem14** [comment]
Same here

**Comment by noxsii** [comment]
same

**Comment by alvarosabu** [comment]
same issue here without the SSR

**Comment by danken00** [comment]
Me too. Did anyone find a workaround for this?

**Comment by braitsch** [comment]
This error will go away if you use v3.0.0-alpha.10 but I still can't get vue-meta to work in a brand new project regardless.

**Comment by numver** [comment]
Same here

**Comment by Rathanavel** [comment]
Same here.

**Comment by renatosaz** [comment]
same here

**Comment by therealbenzene** [comment]
Same here

**Comment by wesly1988** [comment]
same here

**Comment by twentytwo777** [comment]
same here

**Comment by JPaezMin** [comment]
same heve for a vite + vue 3 environment.

**Comment by hsyir** [comment]
same

**Comment by roarkmccolgan** [comment]
SAme

**Comment by sadrahoseini** [comment]
same for vie + vue 3

**Comment by DengSihan** [comment]
same here vite+vue3

**Comment by staszek998** [comment]
I think I've found the source of the problem, as well as the solution:

**TL;DR:** Verify that you've got the correct version of the `vue-meta` package installed in your project.

For some reason, when you run the `npm install vue-meta@next --save` command (as instructed by the docs from [the homepage](https://github.com/nuxt/vue-meta/tree/next)), the NPM installs the `vue-meta` package with the version set to `^3.0.0-alpha.2`. However, when you look at the [Releases](https://github.com/nuxt/vue-meta/releases) page, it's obvious that the newest release is [`3.0.0-alpha.8`](https://github.com/nuxt/vue-meta/releases/tag/v3.0.0-alpha.8). Given this information, when you manually **set the package version in the `package.json` to `^3.0.0-alpha.8` and run the `npm install` command**, the problem goes away; NPM updates the `vue-meta` package, and the `createMetaManager()` function behaves as expected 🙂

---

CC: @pimlie ('cause you seem to maintain [the `README.md` file](https://github.com/nuxt/vue-meta/commits/next/README.md) for the `next` branch)

**Comment by Muhand** [comment]
@staszek998 this fixes the problem. Indeed `npm install vue-meta@next` is installing `alpha.2`

**Comment by Muhand** [comment]
> I think I've found the source of the problem, as well as the solution:
> 
> **TL;DR:** Verify that you've got the correct version of the `vue-meta` package installed in your project.
> 
> For some reason, when you run the `npm install vue-meta@next --save` command (as instructed by the docs from [the homepage](https://github.com/nuxt/vue-meta/tree/next)), the NPM installs the `vue-meta` package with the version set to `^3.0.0-alpha.2`. However, when you look at the [Releases](https://github.com/nuxt/vue-meta/releases) page, it's obvious that the newest release is [`3.0.0-alpha.8`](https://github.com/nuxt/vue-meta/releases/tag/v3.0.0-alpha.8). Given this information, when you manually **set the package version in the `package.json` to `^3.0.0-alpha.8` and run the `npm install` command**, the problem goes away; NPM updates the `vue-meta` package, and the `createMetaManager()` function behaves as expected 🙂
> 
> CC: @pimlie ('cause you seem to maintain [the `README.md` file](https://github.com/nuxt/vue-meta/commits/next/README.md) for the `next` branch)

@staszek998 this fixes the problem. Indeed `npm install vue-meta@next` is installing `alpha.2`
